src/librbd/internal.cc

```diff
@@ -1564,6 +1582,11 @@ int invoke_async_request(ImageCtx *ictx, const std::string& request_type,
           lderr(cct) << "cannot enable object map" << dendl;
           return -EINVAL;
         }
+        if ((ictx->features & RBD_FEATURE_OBJECT_MAP) == 0) {
```

Comment:
I would create the object map after the feature bits have been updated on disk (just in case the feature bit update fails).  I would also create an object map for each existing snapshot to avoid similar "failed to load object map" errors when opening a snapshot.